llama-index-core/llama_index/core/query_engine/omni_modal.py

```diff
@@ -0,0 +1,254 @@
+from typing import Any, Dict, Generic, List, Optional, Sequence
+
+from llama_index.core.base.response.schema import RESPONSE_TYPE, Response
+from llama_index.core.callbacks.base import CallbackManager
+from llama_index.core.callbacks.schema import CBEventType, EventPayload
+from llama_index.core.indices.omni_modal import OmniModalVectorIndexRetriever
+from llama_index.core.embeddings.omni_modal_base import (
+    KD,
+    KQ,
+    Modality,
+    Modalities,
+    OmniModalEmbeddingBundle,
+)
+from llama_index.core.indices.query.base import BaseQueryEngine
+from llama_index.core.indices.query.schema import QueryBundle, QueryType
+from llama_index.core.multi_modal_llms.base import MultiModalLLM
+from llama_index.core.postprocessor.types import BaseNodePostprocessor
+from llama_index.core.prompts import BasePromptTemplate
+from llama_index.core.prompts.default_prompts import DEFAULT_TEXT_QA_PROMPT
+from llama_index.core.prompts.mixin import PromptMixinType
+from llama_index.core.schema import NodeWithScore
+
+
+class OmniModalQueryEngine(BaseQueryEngine, Generic[KD, KQ]):
```

Comment:
Mostly a copy of `MultiModalQueryEngine`. Once we have `OmniModalLLM`, we can generalize this class to other modalities as well.